integration-tests/api/__tests__/store/cart.js

```diff
@@ -480,6 +487,60 @@ describe("/store/carts", () => {
       ])
     })
 
+    it("adding line item to a cart with associated sales channel returns 400", async () => {
+      const api = useApi()
+
+      const response = await api
+        .post(
+          "/store/carts/test-cart-with-sales-channel/line-items",
+          {
+            variant_id: "test-variant-quantity", // variant's product doesn't belong to a sales channel
+            validate_sales_channels: true,
```

Comment:
todo: The `validateSalesChannels` must be part of the context of the server method. It should not be pass through the api. The store api should always validate the sales channels

Reply:
Would it make sense to set this to true by default but still allow the flag to be passed? Since we will allow invalid products to be added from admin.

Reply:
I don’t think so since they are on different domains path admin/store. The store handler always pass true in the context and the admin will pass false in the context. But it is server side